Prose/ProseLib/Sources/ConversationFeature/MessageBar/MessageBar.swift

```diff
@@ -126,22 +158,31 @@ public let messageBarReducer: Reducer<
 // MARK: State
 
 public struct MessageBarState: Equatable {
+  let chatId: JID
+  let loggedInUserJID: JID
+  var typing: [Name]
   var textField: MessageBarTextFieldState
-  var typing: [TypingUser]
   var reactionPicker: ReactionPickerState?
 
   public init(
-    textField: MessageBarTextFieldState,
-    typing: [TypingUser] = []
+    chatId: JID,
+    loggedInUserJID: JID,
+    chatName: Name,
+    message: String = "",
+    typing: [Name] = []
```

Comment:
Same here. `typing` could also be a `Boolean`. Maybe `typingUsers` or `composingUsers` if we want to stick to one terminology?